sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/HardcodedIpAddressBase.cs

```diff
@@ -57,27 +57,27 @@ protected HardcodedIpAddressBase(IAnalyzerConfiguration analyzerConfiguration) :
                 var stringLiteral = (TLiteralExpression)c.Node;
                 var literalValue = GetValueText(stringLiteral);
 
-                if (literalValue == "::" ||
-                    literalValue == "127.0.0.1" ||
-                    !IPAddress.TryParse(literalValue, out var address))
+                var variableName = GetAssignedVariableName(stringLiteral);
+                if (variableName != null &&
+                    IgnoredVariableNames.Any(variableName.Contains))
                 {
                     return;
                 }
 
-                if (address.AddressFamily == AddressFamily.InterNetwork &&
-                    literalValue.Split('.').Length != 4)
+                if (HasAttributes(stringLiteral))
                 {
                     return;
                 }
 
-                var variableName = GetAssignedVariableName(stringLiteral);
-                if (variableName != null &&
-                    IgnoredVariableNames.Any(variableName.Contains))
+                if (literalValue == "::" ||
```

Comment:
I think this is less CPU intensive than `GetAssignedVariableName` and `HasAttributes` which have to visit the syntax tree to search for identifiers and attributes.